swarm/network/simulations/overlay.go

```diff
@@ -19,18 +33,23 @@ import (
 	"github.com/ethereum/go-ethereum/p2p/simulations"
 	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
 	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/state"
+	_ "github.com/ethereum/go-ethereum/swarm/testutil"
```

Comment:
Is this import necessary?

Reply:
This import is needed because that is what makes the whole flags parsing and initializing the `GlogHandler`. As there is no function in that file, but just a `init*()` func, compiler complains, so it needs to be imported this way so that 
1. the `init` function works
2. it can be reused just like this in other tests

Reply:
Well the problem is that flag parsing and processing for logfilters (`vmodule`) as suggested by Felix need to be done in an `init` function. If we want to reuse some code it needs to fit somewhere inside a `init` function. Otherwise everytime we want to use this functionality it needs to be implemented again.

I understand the "...into a file which doesn't actually define any tests." argument but on the other hand it's a simulation, which is inside the testing realm, so I would like to respectfully disagree.

The solution you suggested with using `debug.Setup()` isn't appropriate as it would require a complete binary test setup (`cli` package), for which honestly for this little main (who nobody is using anyway up to now AFAIK) IMHO is a overkill.

The only other solution I can see is implement all the `testutil` functionality in the `init` function in this simulation file. Which makes it not reusable. If this is the preferred way,  please request explicitly, to which I will agree and will delete the `testutil/flag.go` file and put the logic here.
Otherwise please suggest a different solution.

Reply:
I guess my issue here is that we already have functionality in the codebase to manage flags for debug purposes (i.e. the `internal/debug` package), I really don't think it makes sense to introduce another package to effectively do the same thing (i.e. allow a user to specify `--vmodule` on the command line to control logging), and I particularly don't think it should be called `testutil`.

I think the correct solution here is to just use the cli package like all other main packages in the codebase do (it is there exactly for this use case of defining re-usable flags).

If you think that is overkill, then perhaps just add a function in `internal/debug` that does what you want, then here do:

```
func init() {
	debug.<whatever-function-you-define>()
}
```

Reply:
My real goal with this was to introduce a re-usable way for **test cases** and provide kind of a "helper" for tests to reuse. In this context, this PR is a hybrid, as it introduces a test case which is for a main package.

The need arose from the mess of `init()` functions in the `swarm/network/stream/` package (look at all the `*_test.go` files, each with their own `init()` function), which proved to cause confusion in flag parsing, log filtering among other issues.

I will adopt the `init` approach with the `debug.Setup()`